### PR TITLE
Adding ColorSelector [WIP]

### LINKED ISSUE #850: Color property selector
image

Colour text field at simulator should be replaced with the colour selector
_This issue is a part of my [Usability Test](https://drive.google.com/open?id=1igwLx2tc1dAU_75qY2F4u7ZnHf2crcp-)_

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by Shivansh2407**
Fixes #850 Color property selector

#### Describe the changes you have made in this pr -
Here I have added a Color Selector Instead Of the text field . I guess we Do not require the colortorgba function anymore as this selector provides a Hexadecimal value automatically and hence I have commented them ( Will remove them once approved).

### Screenshots of the changes (If any) -
I will provide the screenshots soon

### CODE REVIEW

**Comment by thiefie** [comment]
The color option does not even appear now.
Does this work for you?
Screenshot_2020-02-05_17-29-12

**Comment by Ananthavijay** [comment]
Change the title of PR too

**Comment by Shivansh2407** [comment]
Working On It...Will push some changes soon @thiefie

**Comment by MrBartusek** [comment]
@Shivansh2407 you can just save this pr as a draft or don't make pr at all if you are still working on it

**Comment by nitin10s** [comment]
@Shivansh2407 Are you working on this?

**Comment by Shivansh2407** [comment]
Yes @nitin10s I am working on this. I will open a PR soon

**Comment by nitin10s** [comment]
@Shivansh2407 We should close this one then, right?

**Comment by Shivansh2407** [comment]
I will be adding and squashing the commits in this patch only so I think we should let it be open.

**Comment by MrBartusek** [comment]
@Shivansh2407 any update? are you still working on it?

**Comment by Shivansh2407** [comment]
Yes @MrBartusek I am working on it. Some changes needs to be fixed and then I will update the PR.

File: public/js/modules.js [hunk 1]
```diff
@@ -2296,8 +2296,8 @@ function DigitalLed(x, y, scope = globalScope, color = "Red") {
     this.directionFixed = true;
     this.fixedBitWidth = true;
     this.color = color;
-    var temp = colorToRGBA(this.color)
-    this.actualColor = "rgba(" + temp[0] + "," + temp[1] + "," + temp[2] + "," + 0.8 + ")";
+    //var temp = colorToRGBA(this.color)
+    //this.actualColor = "rgba(" + temp[0] + "," + temp[1] + "," + temp[2] + "," + 0.8 + ")";
```
Comment (satu0king) [inline public/js/modules.js]: I don't think this change is necessary

File: public/js/modules.js [hunk 2]
```diff
@@ -2316,15 +2316,15 @@ DigitalLed.prototype.customSave = function () {
...
         func: "changeColor",
     },
 }
 DigitalLed.prototype.changeColor = function (value) {
     if (validColor(value)) {
         this.color = value;
-        var temp = colorToRGBA(this.color)
-        this.actualColor = "rgba(" + temp[0] + "," + temp[1] + "," + temp[2] + "," + 0.8 + ")";
+        //var temp = colorToRGBA(this.color)
+        //this.actualColor = "rgba(" + temp[0] + "," + temp[1] + "," + temp[2] + "," + 0.8 + ")";
```
Comment (satu0king) [inline public/js/modules.js]: I don't think this change is necessary

File: public/js/modules.js [hunk 3]
```diff
@@ -2316,15 +2316,15 @@ DigitalLed.prototype.customSave = function () {
 DigitalLed.prototype.mutableProperties = {
     "color": {
         name: "Color: ",
-        type: "text",
+        type: "color",
```
Comment (satu0king) [inline public/js/modules.js]: This change is necessary to make change to the UI

**Comment by satu0king** [comment]
Note that, the color picker should pick color, convert to hex and send inside to the function. Therefore this should only be a UI change. I actually prefer having the current option of writing the color name/color hex code as its easier to say I want "red", "green", "blue" rather than picking it from a color picker. Of course some users might need to have the color chosen from a picker.

**Comment by Shivansh2407** [comment]
Ok @satu0king. So then what needs to be done here?

**Comment by satu0king** [comment]
> Ok @satu0king. So then what needs to be done here?

Sorry for delayed response. One way is to give text option just like how it is now and give a color picker version as well. I have seen some applications do this well not able to find a good example now.

**Comment by abstrekt** [comment]
You're working on it right? @Shivansh2407

**Comment by satu0king** [comment]
@Shivansh2407 @abstrekt shall we work on this now?

**Comment by abstrekt** [comment]
> 
> 
> @Shivansh2407 @abstrekt can we work on this now?

what's the progress?

**Comment by satu0king** [comment]
> > @Shivansh2407 @abstrekt can we work on this now?
> 
> what's the progress?

I think we need to decide how the UI will look. As previously discussed in this thread. I think having both color picker + text option is good. Because, it is easier to write "green", "red", "magenta" than picking from a color picker. (I personally find that harder). Also it is important to note that users won't be very picky on their colors as the colors are meant to differentiate rather than aesthetics. The rendering code itself uses any valid CSS color.

@nitin10s any ideas?

**Comment by satu0king** [comment]
Closing this issue as its directed to old UI. Thanks!

**Pull request closed without merging**